tools/mksurfdata_map/src/mkglcmecMod.F90

```diff
@@ -598,29 +599,10 @@ subroutine mkglacier(ldomain, mapfname, datfname, ndiag, zero_out, glac_o)
      ! Global sum of output field -- must multiply by fraction of
      ! output grid that is land as determined by input grid
 
-     sum_fldi = 0.0_r8
-     do ni = 1, ns
-        sum_fldi = sum_fldi + tgridmap%area_src(ni) * tdomain%mask(ni)
-     enddo
-
-     sum_fldo = 0.
-     do no = 1, ns_o
-        sum_fldo = sum_fldo + tgridmap%area_dst(no) * frac_dst(no)
-     end do
-
-     ! -----------------------------------------------------------------
-     ! Error check1
-     ! Compare global sum fld_o to global sum fld_i.
-     ! -----------------------------------------------------------------
-
-     if ( trim(mksrf_gridtype) == 'global') then
-        if ( abs(sum_fldo/sum_fldi-1.) > relerr ) then
-           write (6,*) 'MKGLACIER error: input field not conserved'
-           write (6,'(a30,e20.10)') 'global sum output field = ',sum_fldo
-           write (6,'(a30,e20.10)') 'global sum input  field = ',sum_fldi
-           stop
-        end if
-     end if
+     allocate(mask_r8(tdomain%ns), stat=ier)
+     if (ier/=0) call abort()
+     mask_r8 = tdomain%mask
+     call gridmap_check( tgridmap, mask_r8, frac_dst, subname )
```

Comment:
In the current commit I decided to replace all remaining examples of `Error check 1` with `call gridmap_check` because test 007 was failing with this error in mklanwat.